mutation_cleaner.hh

```diff
@@ -32,19 +32,23 @@ private:
     lw_shared_ptr<worker> _worker_state;
     mutation_application_stats& _app_stats;
     seastar::scheduling_group _scheduling_group;
+    std::function<void(size_t)> _on_space_freed;
 private:
     stop_iteration merge_some(partition_snapshot& snp) noexcept;
     stop_iteration merge_some() noexcept;
     void start_worker();
 public:
     mutation_cleaner_impl(logalloc::region& r, cache_tracker* t, mutation_cleaner* cleaner,
-            mutation_application_stats& app_stats, seastar::scheduling_group sg = seastar::current_scheduling_group())
+            mutation_application_stats& app_stats,
+            seastar::scheduling_group sg = seastar::current_scheduling_group(),
+            std::function<void(size_t)> on_space_freed = nullptr)
```

Comment:
Tangential: we may be able to get rid of byte-granularity accounting with compaction groups.

Let's say we have a largish number N (100-1000) of compaction groups. Currently, we measure the memtable fill bandwidth and memtable outgoing bandwidth by using the memtable byte size, and try to match them by increasing/decreasing flush shares. But if we have a large number of compaction groups then the flush bandwidth can be measured by (occupied size of regions of recently flushed memtables) / (time to flush those memtables). This reacts more slowly to changes, but simplifies the code and removes double accounting.